universal-login-commons/src/core/utils/InitializationHandler.ts

```diff
@@ -0,0 +1,39 @@
+type State<T, B> = {
+  kind: 'Initialized';
```

Comment:
I think names `InitializED` and `FinalizED` are a bit misleading because they imply that the process is already completed which might not be the case